.gitlab-ci.yml

```diff
@@ -0,0 +1,68 @@
+stages:          # List of stages for jobs, and their order of execution
+  - build-docker
+  - test
+
+workflow:
+  rules:
+    # Only run the pipeline if its for a pull request or the default branch
+    - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
+    - if: '$CI_EXTERNAL_PULL_REQUEST_IID != null || $CI_EXTERNAL_PULL_REQUEST_IID != ""'
+
+docker_build:
+  stage: build-docker
+  image: docker:20.10
+  tags:
+    - gpu
+    - docker
+  variables:
+    # For this to work, $CI_COMMIT_REF_SLUG == $CI_DEFAULT_BRANCH
+    DOCKER_TAG: $CI_REGISTRY_IMAGE:$CI_COMMIT_REF_SLUG
```

Comment:
Now that the code and Dockerfile is contained in a single repo, would it make sense to tag the image with the versioneer version in addition to (or instead of) the commit slug?

Reply:
That would prevent the ability to get the last image built for caching, as we would have to determine the last version to pull. It could probably be done, but seems more complicated for no obvious utility.

Reply:
Ah, sorry, I misunderstood what this is doing. I see now that `CI_COMMIT_REF_SLUG` is the abbreviated branch/tag name, and that we're only building the base image here.